### ISSUE TITLE
negative balance

### ISSUE THREAD

**Issue by Boblev999** [connext/nxtp#1407]
Describe the bug
After add liquidity in 3 network, at first I had a negative balance, and then my router completely disappeared from the list?

**To Reproduce**
1. Go to 'https://testnet.amarok.connextscan.io/router/0x6B52b1284f545DB5ac55988fB4643a1eADa33185'
2. 0x6B52b1284f545DB5ac55988fB4643a1eADa33185 my address

**Additional context**
![image](https://user-images.githubusercontent.com/71060373/173459842-04a5fe8e-3001-4dff-8be2-cc8e96d0dd4d.png)
TX Link:
https://kovan.etherscan.io/tx/0x76d992292632f3cd5b2ac3cc2601385216d1eca9d031eb16a7b56920fc232dcd
https://rinkeby.etherscan.io/tx/0xf9826970f4293e6f5dd6ccd0c4818ce018d958630f8f97054f29eca32c1503cc
https://goerli.etherscan.io/tx/0xbcb59ee91ec9d6150d2071aa7148981091dee41927ab5964af1924b4025a92f6

**Comment by just-a-node** [comment]
Subgraph shows router has [negative balance (e.g. kovan)](https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan/graphql?query=query+MyQuery+%7B%0A++router%28id%3A+%220x6b52b1284f545db5ac55988fb4643a1eada33185%22%29+%7B%0A++++assetBalances+%7B%0A++++++amount%0A++++++id%0A++++++asset+%7B%0A++++++++id%0A++++++++canonicalId%0A++++++++canonicalDomain%0A++++++%7D%0A++++%7D%0A++++owner%0A++++isActive%0A++%7D%0A%7D).

**Comment by rhlsthrm** [comment]
Looks like it's an error in our mapping then.

**Comment by just-a-node** [comment]
Quite a few [negative balances](https://api.thegraph.com/subgraphs/name/connext/nxtp-amarok-runtime-v0-kovan/graphql?query=query+MyQuery+%7B%0A++routers+%7B%0A++++id%0A++++assetBalances%28where%3A+%7Bamount_lt%3A+%220%22%7D%29+%7B%0A++++++amount%0A++++++asset+%7B%0A++++++++id%0A++++++++adoptedAsset%0A++++++%7D%0A++++%7D%0A++%7D%0A%7D). Team will investigate further.

**Comment by jakekidd** [comment]
We've determined that the cause of this issue is likely due to a prior mishandling of the slow-liquidity route on-chain - a bug that was fixed >13 days ago. However, the event history still has mishandled (or inaccurate) events from before this issue was corrected.

We've added a redundant check in the subgraph mappings to achieve backwards compatibility and correct inaccurate events leading to poor state updates.

See #1417
